doc/user/metrics/README.md

```diff
@@ -74,15 +74,129 @@ In Kubernetes clusters where [OwnerReferencesPermissionEnforcement][ownerref-per
   - update
 ...
 ```
-
 ### Custom resource specific metrics
 
 By default operator will expose info metrics based on the number of the current instances of an operator's custom resources in the cluster. It leverages [kube-state-metrics][ksm] as a library to generate those metrics. Metrics initialization lives in the `cmd/manager/main.go` file of the operator in the `serveCRMetrics` function. Its arguments are a custom resource's group, version, and kind to generate the metrics. The metrics are served on `0.0.0.0:8686/metrics` by default. To modify the exposed metrics port number, change the `operatorMetricsPort` variable at the top of the `cmd/manager/main.go` file in the generated operator.
 
+### Custom Metrics using Operator-SDK
+To implement custom metric collection using the `metrics package` provided by SDK, follow the steps outlined below:
+1. Update the Operator: Update the operator to provide sufficient permisisons so that the required resources for prometheus can be created.
+2. Custom metric collection: Implement the logic for custom metric variables defined for your operator and register the specific metrics with prometheus. 
+3. Expose metrics: Utilize the functions provided by SDK to expose metrics and start the metrics server.
+
+#### Update the Operator
+
+Update the operator and provide sufficient permissions to allow creation of Prometheus resources. The required permissions are:
+1. `Service Monitor` : `get` and `create`
+2. `roles` and `rolebindings` : `get`, `create`, `delete`, `update`, `list`
+
+The below is a snippet from roles.yaml:
+```yaml
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - get
+  - create
+  - delete
+  - update
+  - list
+  - watch
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  verbs:
+  - get
+  - create
```

Comment:
The following part is already present in the default roles added in the Role.yaml, so should it be really required to be added here? Should not better just add what is required to do in order to make it work?  

```
- apiGroups:
  - monitoring.coreos.com
  resources:
  - servicemonitors
  verbs:
  - get
  - create
```
IMHO: we could remove this part.